### PR TITLE
Better UI for index page in the embedded web server

### PR DESCRIPTION
**Pull request by nivoc**
Possible Simple Layout

### CODE REVIEW

File: src/jaguar.toit [hunk 1]
```diff
@@ -282,22 +282,89 @@ handle_browser_request request/http.Request writer/http.ResponseWriter -> none:
...
-            <h2 class=help>Monitor the serial port console using <a href="https://github.com/toitlang/jaguar"><code>jag monitor</code></a></h2>
-            <p>
-              <img src="$CHIP_IMAGE" alt="Picture of an embedded device" width=200 />
-            </p>
-          </body>
-        </html>
+<html>
+  <head>
+    <title>$device.name (Jaguar device)</title>
+    <style>
```
Comment (kasperl) [inline src/jaguar.toit:288]: I'd move the style part into the style.css file that is served separately.

File: src/jaguar.toit [hunk 2]
```diff
@@ -282,22 +282,89 @@ handle_browser_request request/http.Request writer/http.ResponseWriter -> none:
...
-            <h2>Uptime: $(Duration --s=Time.monotonic_us / Duration.MICROSECONDS_PER_SECOND)</h2>
-            <h2>Toit SDK version: $vm_sdk_version</h2>
-            <h2 class=help>Run code on this device using <a href="https://github.com/toitlang/jaguar"><code>jag run</code></a></h2>
-            <h2 class=help>Monitor the serial port console using <a href="https://github.com/toitlang/jaguar"><code>jag monitor</code></a></h2>
-            <p>
-              <img src="$CHIP_IMAGE" alt="Picture of an embedded device" width=200 />
-            </p>
-          </body>
-        </html>
+<html>
```
Comment (kasperl) [inline src/jaguar.toit:285]: Toit tries to be smart about stripping the leading indentation. Maybe keep it indented, so it flows naturally with the source code?

File: src/jaguar.toit [hunk 3]
```diff
@@ -282,22 +282,89 @@ handle_browser_request request/http.Request writer/http.ResponseWriter -> none:
...
+          background: #fff;
+          border-radius: 15px;
+          box-shadow: rgb(255, 255, 255) 0px 0px 0px 0px, rgba(51, 65, 85, 0.1) 0px 0px 0px 1px, rgba(0, 0, 0, 0.05) 0px 20px 25px -5px, rgba(0, 0, 0, 0.05) 0px 8px 10px -6px;
+          box-sizing: border-box;
+          color: #444;
+          display: block;
+          line-height: 24px;
+          padding: 12px;
+          width: 390px;
+          margin:auto;
```
Comment (kasperl) [inline src/jaguar.toit:318]:
```suggestion
          margin: auto;
```

File: src/jaguar.toit [hunk 4]
```diff
@@ -282,22 +282,89 @@ handle_browser_request request/http.Request writer/http.ResponseWriter -> none:
...
+          margin-top: 80px;
+          padding-left:20px;
+        }
+        .icon {
+          rotate: 45deg;
+          padding-top: 20px;
+          color: #62BBF3;
+          position: relative;
+          top:5px;
```
Comment (kasperl) [inline src/jaguar.toit:327]:
```suggestion
          top: 5px;
```

File: src/jaguar.toit [hunk 5]
```diff
@@ -282,22 +282,89 @@ handle_browser_request request/http.Request writer/http.ResponseWriter -> none:
...
+          left:5px;
```
Comment (kasperl) [inline src/jaguar.toit:328]:
```suggestion
          left: 5px;
```

File: src/jaguar.toit [hunk 6]
```diff
@@ -282,22 +282,89 @@ handle_browser_request request/http.Request writer/http.ResponseWriter -> none:
...
+          text-align:center
```
Comment (kasperl) [inline src/jaguar.toit:329]:
```suggestion
          text-align: center
```

File: src/jaguar.toit [hunk 7]
```diff
@@ -282,22 +282,89 @@ handle_browser_request request/http.Request writer/http.ResponseWriter -> none:
...
-            <p>
-              <img src="$CHIP_IMAGE" alt="Picture of an embedded device" width=200 />
-            </p>
-          </body>
-        </html>
+<html>
+  <head>
+    <title>$device.name (Jaguar device)</title>
+    <style>
+        body {
```
Comment (kasperl) [inline src/jaguar.toit:289]: I'd stick with 2-space indent consistently.

File: src/jaguar.toit [hunk 8]
```diff
@@ -282,22 +282,89 @@ handle_browser_request request/http.Request writer/http.ResponseWriter -> none:
...
+          font-size: 28px;
+          margin-top: 0px;
+          margin-bottom: 5px;
+        }
+        a {
+          text-decoration: none;
+          color: #333;
+        }
+        p {
+          margin:0
```
Comment (kasperl) [inline src/jaguar.toit:304]:
```suggestion
          margin: 0px;
```

File: src/jaguar.toit [hunk 9]
```diff
@@ -282,22 +282,89 @@ handle_browser_request request/http.Request writer/http.ResponseWriter -> none:
...
-        </html>
+<html>
+  <head>
+    <title>$device.name (Jaguar device)</title>
+    <style>
+        body {
+            background-color: #F5F5F5;
+        }
+        h1 {
+          font-family: ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial;
```
Comment (kasperl) [inline src/jaguar.toit:293]: Are all these necessary?

File: src/jaguar.toit [hunk 10]
```diff
@@ -282,22 +282,89 @@ handle_browser_request request/http.Request writer/http.ResponseWriter -> none:
...
+        a {
+          text-decoration: none;
+          color: #333;
+        }
+        p {
+          margin:0
+        }
+        .box {
+          font-family: monospace;
+          border:none;
```
Comment (kasperl) [inline src/jaguar.toit:308]:
```suggestion
          border: none;
```

File: src/jaguar.toit [hunk 11]
```diff
@@ -282,22 +282,89 @@ handle_browser_request request/http.Request writer/http.ResponseWriter -> none:
...
+          box-shadow: rgb(255, 255, 255) 0px 0px 0px 0px, rgba(51, 65, 85, 0.1) 0px 0px 0px 1px, rgba(0, 0, 0, 0.05) 0px 20px 25px -5px, rgba(0, 0, 0, 0.05) 0px 8px 10px -6px;
+          box-sizing: border-box;
+          color: #444;
+          display: block;
+          line-height: 24px;
+          padding: 12px;
+          width: 390px;
+          margin:auto;
+          margin-top: 80px;
+          padding-left:20px;
```
Comment (kasperl) [inline src/jaguar.toit:320]:
```suggestion
          padding-left: 20px;
```

File: src/jaguar.toit [hunk 12]
```diff
@@ -282,22 +282,89 @@ handle_browser_request request/http.Request writer/http.ResponseWriter -> none:
...
+          rotate: 45deg;
+          padding-top: 20px;
+          color: #62BBF3;
+          position: relative;
+          top:5px;
+          left:5px;
+          text-align:center
+        }
+        pre {
+          margin: 0
```
Comment (kasperl) [inline src/jaguar.toit:332]:
```suggestion
          margin: 0px;
```

File: src/jaguar.toit [hunk 13]
```diff
@@ -282,22 +282,89 @@ handle_browser_request request/http.Request writer/http.ResponseWriter -> none:
...
+    <div class="box"> 
+        <div class="center">
+            <svg width="90" style="" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" class="icon">
+            <path stroke-linecap="round" stroke-linejoin="round" d="M8.25 3v1.5M4.5 8.25H3m18 0h-1.5M4.5 12H3m18 0h-1.5m-15 3.75H3m18 0h-1.5M8.25 19.5V21M12 3v1.5m0 15V21m3.75-18v1.5m0 15V21m-9-1.5h10.5a2.25 2.25 0 002.25-2.25V6.75a2.25 2.25 0 00-2.25-2.25H6.75A2.25 2.25 0 004.5 6.75v10.5a2.25 2.25 0 002.25 2.25zm.75-12h9v9h-9v-9z" />
+          </svg>
+        </div>
+        <br>        
+        <h1>$device.name</h1>
+        <div class="center">Jaguar device</div>
+        <br>
```
Comment (kasperl) [inline src/jaguar.toit:349]: Maybe avoid this by changing padding on the "center" style and call it something else?

File: src/jaguar.toit [hunk 14]
```diff
@@ -282,22 +282,89 @@ handle_browser_request request/http.Request writer/http.ResponseWriter -> none:
...
+<pre>
+=============================================
+= UPTIME: <b>$(Duration --s=Time.monotonic_us / Duration.MICROSECONDS_PER_SECOND)</b>
```
Comment (kasperl) [inline src/jaguar.toit:352]: Compute this before the string literal that starts with """, so it looks better here? 

```
uptime ::= Duration --s=Time.monotonic_us / Duration.MICROSECONDS_PER_SECOND
writer.write """
...
```

and then this becomes:

```suggestion
= UPTIME: <b>$uptime</b>                            
```

**Review by kasperl — COMMENTED** [review]
Thank you for working on this! Here are a few comments:

**Comment by nivoc** [comment]
Sry for the delay. I was knocked out for 2 days. I worked in the comments and improved the layout a bit. 
image

File: src/jaguar.toit (hunk 1 above)
Comment (nivoc) [reply src/jaguar.toit]: changed with [75986fc](https://github.com/toitlang/jaguar/pull/226/commits/75986fc47f932a95519700856d385a240149c0a9)

File: src/jaguar.toit (hunk 2 above)
Comment (nivoc) [reply src/jaguar.toit]: done [75986fc](https://github.com/toitlang/jaguar/pull/226/commits/75986fc47f932a95519700856d385a240149c0a9)

File: src/jaguar.toit (hunk 3 above)
Comment (nivoc) [reply src/jaguar.toit]: done [75986fc](https://github.com/toitlang/jaguar/pull/226/commits/75986fc47f932a95519700856d385a240149c0a9)

File: src/jaguar.toit (hunk 4 above)
Comment (nivoc) [reply src/jaguar.toit]: done [75986fc](https://github.com/toitlang/jaguar/pull/226/commits/75986fc47f932a95519700856d385a240149c0a9)

File: src/jaguar.toit (hunk 5 above)
Comment (nivoc) [reply src/jaguar.toit]: done [75986fc](https://github.com/toitlang/jaguar/pull/226/commits/75986fc47f932a95519700856d385a240149c0a9)

File: src/jaguar.toit (hunk 10 above)
Comment (nivoc) [reply src/jaguar.toit:337]: done [75986fc](https://github.com/toitlang/jaguar/pull/226/commits/75986fc47f932a95519700856d385a240149c0a9)

File: src/jaguar.toit (hunk 11 above)
Comment (nivoc) [reply src/jaguar.toit:350]: done [75986fc](https://github.com/toitlang/jaguar/pull/226/commits/75986fc47f932a95519700856d385a240149c0a9)

File: src/jaguar.toit (hunk 12 above)
Comment (nivoc) [reply src/jaguar.toit]: done [75986fc](https://github.com/toitlang/jaguar/pull/226/commits/75986fc47f932a95519700856d385a240149c0a9)

File: src/jaguar.toit (hunk 13 above)
Comment (nivoc) [reply src/jaguar.toit]: Not 100% sure. The new version has no center anymote.

File: src/jaguar.toit (hunk 14 above)
Comment (nivoc) [reply src/jaguar.toit]: done [75986fc](https://github.com/toitlang/jaguar/pull/226/commits/75986fc47f932a95519700856d385a240149c0a9)

**Review by erikcorry — APPROVED** [review]

**Comment by erikcorry** [comment]
Thanks for this!

**Pull request merged**